clients/src/main/java/org/apache/kafka/common/metrics/Sensor.java

```diff
@@ -53,12 +54,20 @@
     private final Object metricLock;
 
     private static class StatAndConfig {
-        public final Stat stat;
-        public final MetricConfig config;
+        private final Stat stat;
+        private final Supplier<MetricConfig> configSupplier;
 
-        StatAndConfig(Stat stat, MetricConfig config) {
+        StatAndConfig(Stat stat, Supplier<MetricConfig> configSupplier) {
             this.stat = stat;
-            this.config = config;
+            this.configSupplier = configSupplier;
+        }
+
+        public Stat stat() {
+            return this.stat;
+        }
+
+        public MetricConfig config() {
+            return this.configSupplier.get();
```

Comment:
I guess if you are removing `this.` above, you could remove it here as well for consistency.